### ISSUE TITLE
`preferLocal` fails for globally installed CLI programs since v3.0.0

### ISSUE THREAD

**Issue by yuezk** [sindresorhus/execa#404]
### Description

I installed `protractor` locally and globally. When I tried to use `preferLocal: true` to use the locally installed `protractor`, it always uses the globally installed `protractor`.

### Example

```js
(async () => {
	const { stdout } = await execa('which', ['protractor'], { preferLocal: true });
	console.log(stdout);
})();
```
I dived into the code and found that this seems to be an issue of `npm-run-path`, and it was introduced with https://github.com/sindresorhus/npm-run-path/pull/8, in which, the `execPath` was prepend to the `PATH`. As the globally installed cli programs also reside in the `execPath` and the `execPath` has a higher priority, the `preferLocal` option no longer work.

**Comment by ehmicky** [comment]
@yuezk, you're absolutely right, thanks a lot for pointing this out.

I have submitted a fix in Execa's dependency: https://github.com/sindresorhus/npm-run-path/pull/10

**Comment by ehmicky** [comment]
This should be fixed by `npm-run-path@4.0.1` which should be used by `execa` if you regenerate your `package-lock.json` or `yarn.lock`.
